core/src/mindustry/world/blocks/power/PowerNode.java

```diff
@@ -357,9 +357,10 @@ public static boolean insulated(int x, int y, int x2, int y2){
         public void placed(){
             if(net.client() || power.links.size > 0) return;
 
+            var player = Groups.player.find(p -> p.name.equals(lastAccessed)); //player who built this block
```

Comment:
This is terrible, there must be a better way of fetching placement.

Reply:
Why not just pass the builder to the placed function? Its only called in once place so it shouldn't be that hard (though it would probably break mods)

Reply:
That's fine, and breaking isn't an issue as the private branch breaks absolutely everything anyway.

Reply:
> the private branch breaks absolutely everything anyway

Stop, you're making me scared. When I merge the average update, there are ~10 conflicts to merge manually, when the v7 pre alpha came out I had over 100 files with conflicts...

Reply:
> you're making me scared

~~maybe you should be~~

![image](https://user-images.githubusercontent.com/10100000/160491626-ceb5f7af-4126-4fa4-b121-c265fc70c454.png)

Reply:
> ~maybe you should be~

Oh good, formations cause me so much pain

~i think this might actually crash intellij~